test/system/060-mount.bats

```diff
@@ -321,4 +321,83 @@ EOF
+    ln -r -s $rel_dir/data $rel_dir/link
+    # Absolute symlink
+    echo "$abs_random_host" > $abs_dir/data
+    ln -s $abs_dir/data $abs_dir/link
+
+    dockerfile=$PODMAN_TMPDIR/Dockerfile
+    cat >$dockerfile <<EOF
+FROM $IMAGE
+RUN echo $random_img > /tmp/data
+EOF
+
+    img="localhost/preserve:symlinks"
+    run_podman build -t $img -f $dockerfile
+
+    link_path="/tmp/link"
+    create_path="/tmp/i/do/not/exist/link"
+
+    tests="
+0 | bind | $rel_dir/link | /tmp/link |  | /tmp/link | $rel_random_host | $link_path | bind mount relative symlink: mounts target from the host
+0 | bind | $abs_dir/link | /tmp/link |  | /tmp/link | $abs_random_host | $link_path | bind mount absolute symlink: mounts target from the host
+0 | glob | $rel_dir/lin* | /tmp/     |  | /tmp/link | $rel_random_host | $link_path | glob mount relative symlink: mounts target from the host
+0 | glob | $abs_dir/lin* | /tmp/     |  | /tmp/link | $abs_random_host | $link_path | glob mount absolute symlink: mounts target from the host
+0 | glob | $rel_dir/*    | /tmp/     |  | /tmp/link | $rel_random_host | $link_path | glob mount entire directory: mounts relative target from the host
+0 | glob | $abs_dir/*    | /tmp/     |  | /tmp/link | $abs_random_host | $link_path | glob mount entire directory: mounts absolute target from the host
+0 | bind | $rel_dir/link | /tmp/link | ,no-dereference | '/tmp/link' -> 'data' | $random_img      | $link_path | no_deref: bind mount relative symlink: points to file on the image
+0 | glob | $rel_dir/lin* | /tmp/     | ,no-dereference | '/tmp/link' -> 'data' | $random_img      | $link_path | no_deref: glob mount relative symlink: points to file on the image
+0 | bind | $rel_dir/     | /tmp/     | ,no-dereference | '/tmp/link' -> 'data' | $rel_random_host | $link_path | no_deref: bind mount the entire directory: preserves symlink automatically
+0 | glob | $rel_dir/*    | /tmp/     | ,no-dereference | '/tmp/link' -> 'data' | $rel_random_host | $link_path | no_deref: glob mount the entire directory: preserves symlink automatically
+1 | bind | $abs_dir/link | /tmp/link | ,no-dereference | '/tmp/link' -> '$abs_dir/data' | cat: can't open '/tmp/link': No such file or directory | $link_path | bind mount *preserved* absolute symlink: now points to a non-existent file on the container
+1 | glob | $abs_dir/lin* | /tmp/     | ,no-dereference | '/tmp/link' -> '$abs_dir/data' | cat: can't open '/tmp/link': No such file or directory | $link_path | glob mount *preserved* absolute symlink: now points to a non-existent file on the container
+0 | bind | $rel_dir/link | $create_path |  | $create_path | $rel_random_host | $create_path | bind mount relative symlink: creates dirs and mounts target from the host
+1 | bind | $rel_dir/link | $create_path | ,no-dereference | '$create_path' -> 'data' | cat: can't open '$create_path': No such file or directory | $create_path | no_deref: bind mount relative symlink: creates dirs and mounts target from the host
```

Comment:
Impressive test suite. Large matrix-type (as opposed to open-ended) tables make me uncomfortable, because it's hard to confirm that all permutations are being covered. For these cases I prefer
```bash
    for mount_type in bind glob; do
        for mount_src in rel abs; do
            for mount_opts in "" ",no-dereference"; do
                  ....compute everything else from those, then test
```
I'm trying to review, and trying to see if the missing table test cases are important ones, and I don't think so. Hence, LGTM2. However, I'm adding this to my rainy-day todo list, to see if I can cleanly rewrite it using a for loop.

Nice work.

(If for any reason you need to repush, there's a few misspellings that might want correction, but none are blockers).

Reply:
@edsantiago, we missed the 4.8 deadline, so we can polish more if needed.  Feel free to block/propose etc.

Reply:
FWIW, this is the missing test from the matrix:
```
1 | glob | $abs_dir/*    | /tmp/        | ,no-dereference | '/tmp/link' -> '$abs_dir/data' | cat: can't open '/tmp/link': No such file or directory    | $link_path   | FIXME THIS WAS MISSING FROM ORIGINAL TEST
```
I've got a WIP that covers the conditions using logic, not a table. It also adds more conditions that I feel are worth testing. Will try to submit it for review tomorrow, but if not, there's no hurry.